dictionary/lib/dictionary/impl.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Dictionary.Impl do
+
+    use Agent
+
+    @me WordListAgent
+
+    def random_word() do
```

Comment:
# -5

The problem here is that you are fetching the entire word list from the agent, and then picking a word. If instead you had

~~~ elixir
Agent.get(@me, &Enum.random/1)
~~~

that would would be done in the agent process, and only a single word would be returned.